contracts/adapters/FlexibleLeverageStrategyAdapter.sol

```diff
@@ -184,30 +217,36 @@ contract FlexibleLeverageStrategyAdapter is BaseAdapter {
 
     /* ============ State Variables ============ */
 
-    ContractSettings internal strategy;             // Struct of contracts used in the strategy (SetToken, price oracles, leverage module etc)
-    MethodologySettings internal methodology;       // Struct containing methodology parameters
-    ExecutionSettings internal execution;           // Struct containing execution parameters
-    IncentiveSettings internal incentive;           // Struct containing incentive parameters for ripcord
-    uint256 public twapLeverageRatio;               // Stored leverage ratio to keep track of target between TWAP rebalances
-    uint256 public lastTradeTimestamp;              // Last rebalance timestamp. Must be past rebalance interval to rebalance
+    ContractSettings internal strategy;                             // Struct of contracts used in the strategy (SetToken, price oracles, leverage module etc)
+    MethodologySettings internal methodology;                       // Struct containing methodology parameters
+    ExecutionSettings internal execution;                           // Struct containing execution parameters
+    mapping(string => ExchangeSettings) internal exchangeSettings;  // Mapping from exchange name to exchange settings
+    IncentiveSettings internal incentive;                           // Struct containing incentive parameters for ripcord
+    string[] public enabledExchanges;                               // Array containing enabled exchanges
+    uint256 public twapLeverageRatio;                               // Stored leverage ratio to keep track of target between TWAP rebalances
+    uint256 public globalLastTradeTimestamp;                        // Last rebalance timestamp. Must be past rebalance interval to rebalance
```

Comment:
More specificity on that this, current timestamp must be greater than this variable by the rebalance interval to rebalance